### PR TITLE
[ONNX] Add binary_cross_entropy_with_logits op to ONNX opset version 12

### PR DESCRIPTION
**Pull request by hwangdeyu**
Fixes #{#47997}
Exporting the operator binary_cross_entropy_with_logits to ONNX opset version 12.

### CODE REVIEW

**Comment by BowenBao** [comment]
please rebase with onnx_ms_1 to resolve CI issues.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -4809,6 +4817,68 @@ def forward(self, input, target):
...
+        self._bce_logits_loss(x, y, weight, pos_weight)
+
+    def _bce_logits_loss(self, x, y, weight=None, pos_weight=None):
+        class BCEWithLogitsLossNone(torch.nn.Module):
+            def forward(self, input, target):
+                return torch.nn.functional.binary_cross_entropy_with_logits(input, target, reduction='none')
+
+        self.run_test(BCEWithLogitsLossNone(), input=(x, y))
+
+        class BCEWithLogitsLossNoneWeights(torch.nn.Module):
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:4848]: Duplicate test? weight/pos_weight being None are tested by above.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -4809,6 +4817,68 @@ def forward(self, input, target):
...
+        y = torch.randn(2, 3, 5, 7)
+        self._bce_logits_loss(x, y)
+
+        x = torch.FloatTensor([[-0.4089, -1.2471, 0.5907], [-0.4897, -0.8267, -0.7349], [0.5241, -0.1246, -0.4751]])
+        y = torch.FloatTensor([[0, 1, 1], [0, 0, 1], [1, 0, 1]])
+        self._bce_logits_loss(x, y)
+
+        x = torch.randn(3, 3, requires_grad=True)
+        y = torch.empty(3, 3).random_(2)
+        weight = torch.tensor([3], dtype=torch.float)
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:4837]: It is mentioned that weight should have shape (batch,) and pos_weight (num_of_class,). Consider adjusting the test input shape accordingly?

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (hwangdeyu) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:4859]: Thanks for this useful comments, I have removed those duplicate tests.

File: torch/onnx/symbolic_opset12.py [hunk 3]
```diff
@@ -51,6 +51,34 @@ def nll_loss2d(g, self, target, weight, reduction, ignore_index):
...
+
+    reduction = sym_help._maybe_get_const(reduction, 'i')
+    if reduction == 0:
+        return output
+    elif reduction == 1:
+        return g.op("ReduceMean", output)
+    elif reduction == 2:
+        return g.op("ReduceSum", output)
+    else:
+        return sym_help._onnx_unsupported("binary_cross_entropy_with_logits")
```
[the remark this answers is not in the document]
Comment (hwangdeyu) [reply torch/onnx/symbolic_opset12.py]: ok, have changed the error string and now the error message as follow:
>RuntimeError: Unsupported: ONNX export of operator binary_cross_entropy_with_logits with reduction other than none, mean, or sum. Please open a bug to request ONNX export support for the missing operator.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 2 above)
Comment (hwangdeyu) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Yes, fixed it and added a slight change to test.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -1293,6 +1293,14 @@ def forward(self, x):
                       dynamic_axes={'input_1': [0, 1, 2],
                                     'output_1': [0, 1, 2]})
 
+    def test_square(self):
```
[the remark this answers is not in the document]
Comment (hwangdeyu) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Thanks, it solved the CI problem perfectly.

File: torch/onnx/symbolic_opset12.py [hunk 5]
```diff
@@ -52,6 +52,34 @@ def nll_loss2d(g, self, target, weight, reduction, ignore_index):
...
+@parse_args('v', 'v', 'v', 'v', 'i')
+def binary_cross_entropy_with_logits(g, input, target, weight, pos_weight, reduction):
+    from torch.onnx.symbolic_opset9 import sigmoid, log, sub, neg, mul, add
+    p = g.op("Constant", value_t=torch.tensor([1]))
+    sig_x = sigmoid(g, input)
+    log_sig_x = log(g, sig_x)
+    sub_1_x = sub(g, p, sig_x)
+    sub_1_y = sub(g, p, target)
+    log_1_x = log(g, sub_1_x)
+    if pos_weight is None or pos_weight.node().mustBeNone():
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset12.py:64]: could you update `_is_none` (with the None check) in `symbolic_helper.py`, and use that here?

File: torch/onnx/symbolic_opset12.py [hunk 6]
```diff
@@ -52,6 +52,34 @@ def nll_loss2d(g, self, target, weight, reduction, ignore_index):
...
+        output = neg(g, add(g, mul(g, target, log_sig_x), mul(g, sub_1_y, log_1_x)))
+    else:
+        output = neg(g, add(g, mul(g, mul(g, target, log_sig_x), pos_weight), mul(g, sub_1_y, log_1_x)))
+
+    if weight is not None and not sym_help._is_none(weight):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset12.py:69]: ditto

**Review by BowenBao — APPROVED** [review]
LGTM, thanks! minor comments with improving helper function.

File: torch/onnx/symbolic_opset12.py (hunk 5 above)
Comment (hwangdeyu) [reply torch/onnx/symbolic_opset12.py]: Thanks! updated it.

**Pull request merged**